# Intent: reliability fix from the Fable review: self.robot.get_head_angles read both servos over the SCS UART (10 ms timeout each) while the main task held the LVGL lock. It now returns the animated angle the motion code already tracks (the target the head is moving to or holds), with no bus read. The call still goes through the MCP round trip, so the host's 'does the unit answer' check is unchanged.
# Upstream: m5stack/StackChan firmware/main/hal/hal_mcp.cpp (MIT). Applies after 039 with --unidiff-zero.
# Check after a flash: get_head_angles returns within a few ms during a reply; a head moved by hand while torque is off is reported at its last commanded angle.
diff --git a/firmware/main/hal/hal_mcp.cpp b/firmware/main/hal/hal_mcp.cpp
index f58823f..8ee8206 100644
--- a/firmware/main/hal/hal_mcp.cpp
+++ b/firmware/main/hal/hal_mcp.cpp
@@ -181,2 +181,2 @@ void Hal::xiaozhi_mcp_init()
-                           int current_yaw   = motion.yawServo().getCurrentAngle() / 10;
-                           int current_pitch = motion.pitchServo().getCurrentAngle() / 10;
+                           int current_yaw   = motion.yawServo().Servo::getCurrentAngle() / 10;
+                           int current_pitch = motion.pitchServo().Servo::getCurrentAngle() / 10;
